internal/cluster/registrar.go

```diff
@@ -69,27 +70,23 @@ func (r *registrar) registerCluster(ctx context.Context, cluster *common.Cluster
 		tx = common.LoadTableFromDB(tx, tableName)
 	}
 
+	fmt.Printf("YEV - creating cluster %s\n", cluster.ID)
 	if err := tx.Create(cluster).Error; err != nil {
 		r.log.Errorf("Error registering cluster %s", cluster.Name)
 		return err
 	}
 
-	if err := tx.Commit().Error; err != nil {
+	err := common.CreateInfraEnvForCluster(tx, cluster)
```

Comment:
Why not create it in the inventory.go level? We will probably use the same code for v2 no? so why not create it with internal function in inventory.go and with v1 you will create it and with v2 you will not.

Reply:
if we create it in inventory.go, then we must define a transaction at that level and pass it down via cluster API. I prefer to create it directly during registrar, without changing inventory code. In V2 InfraEnv for cluster will be created by a InfraEnv API, and not as part of Cluster creation.

Reply:
in addition , using registrar also covers the cases of Day2 cluster (at least until @nmagnezi  delivers) and covers the unregister of the infra-env

Reply:
What is the problem creating the transaction in inventory.go and how will you handle v2 cluster create?

Reply:
as agreed, leaving the creation/destruction of the InfraEnv in the registar, passing a flag that defines it is a v1/v2 implementation (create or not infra env)
This way we have only one place where we create/delete infra-env for V1, and it will take care of day1 cluster, day2 cluster and cluster garbage collector